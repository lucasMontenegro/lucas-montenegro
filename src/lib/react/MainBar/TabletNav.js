import { makeStyles } from "@material-ui/core/styles"
import React from "react"
import List from "@material-ui/core/List"
import { useRoutingContext } from "lib/react/routing/context"
import Tooltip from "@material-ui/core/Tooltip"
import Link from "lib/react/links/Link"
import ListItem from "@material-ui/core/ListItem"
import ListItemIcon from "@material-ui/core/ListItemIcon"
const useStyles = makeStyles(theme => ({
  li: {
    color: theme.palette.primary.main,
  },
  activeLi: {
    color: theme.palette.type === `dark` ? `#33ccd7` : `#017982`,
    backgroundColor: theme.palette.type === `dark` ? `#272727` : `#f1f1f1`,
    borderLeft: `${theme.spacing(0.5)}px solid transparent`,
    borderRight: `${theme.spacing(0.5)}px solid currentColor`,
  },
  anchor: {
    display: `block`,
    color: `inherit`,
  },
  muiLi: {
    justifyContent: `center`,
  },
  liIcon: {
    color: `inherit`,
    minWidth: `initial`,
  },
}), { name: `lib-react-main_bar-tablet_nav` })
export default function TabletNav ({ t }) {
  const classes = useStyles()
  return (
    <nav
      aria-label={t({
        en: () => `Navigation links`,
        es: () => `Links de navegación`,
      })}
    >
      <List disablePadding>
        {useRoutingContext().clientLinks.map(link => (
          <li
            key={link.clientName}
            className={link.active ? `${classes.li} ${classes.activeLi}` : classes.li}
            aria-label={link.active ? t({
              en: () => `Current page`,
              es: () => `Página actual`,
            }) : undefined}
          >
            <Tooltip title={t(link.render.text)}>
              <Link className={classes.anchor} to={link.location}>
                <ListItem component="div" className={classes.muiLi}>
                  <ListItemIcon className={classes.liIcon}>
                    <link.render.Icon t={t} />
                  </ListItemIcon>
                </ListItem>
              </Link>
            </Tooltip>
          </li>
        ))}
      </List>
    </nav>
  )
}