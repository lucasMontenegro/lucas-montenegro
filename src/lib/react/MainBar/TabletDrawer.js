import { makeStyles } from "@material-ui/core/styles"
import useTranslation from "lib/react/useTranslation"
import React from "react"
import Drawer from "@material-ui/core/Drawer"
import List from "@material-ui/core/List"
import Tooltip from "@material-ui/core/Tooltip"
import ListItem from "@material-ui/core/ListItem"
import { useAppDrawerOpener } from "lib/react/AppDrawer"
//import { useDashboardOpener } from "lib/react/Dashboard"
import ListItemIcon from "@material-ui/core/ListItemIcon"
import { FontAwesomeIcon } from "lib/react/fontAwesome"
import Divider from "@material-ui/core/Divider"
import TabletNav from "./TabletNav"
const useStyles = makeStyles(theme => ({
  drawerPaper: {
    width: theme.spacing(10) + 1,
  },
  muiLi: {
    justifyContent: `center`,
  },
  liIcon: {
    fontSize: theme.spacing(3),
    minWidth: `initial`,
  },
}), { name: `lib-react-main_bar-tablet_drawer` })
export default function TabletDrawer () {
  const classes = useStyles()
  const t = useTranslation()
  return (
    <Drawer classes={{ paper: classes.drawerPaper }} variant="permanent">
      <List disablePadding>
        <li>
          <Tooltip
            title={t({
              en: () => `Open dashboard`,
              es: () => `Abrir panel de control`,
            })}
          >
            <ListItem
              className={classes.muiLi}
              button
              aria-controls="lib-react-app_drawer"
              onClick={useAppDrawerOpener()}
            >
            {/*
              aria-controls="lib-react-dashboard"
              onClick={useDashboardOpener()}
            >
            */}
              <ListItemIcon className={classes.liIcon}>
                <FontAwesomeIcon icon={[`fas`, `bars`]} />
              </ListItemIcon>
            </ListItem>
          </Tooltip>
        </li>
        <Divider />
        <li><TabletNav t={t} /></li>
      </List>
    </Drawer>
  )
}