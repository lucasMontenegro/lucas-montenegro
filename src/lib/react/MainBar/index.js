import { makeStyles } from "@material-ui/core/styles"
import React from "react"
import useResponsiveLayout from "lib/react/useResponsiveLayout"
import TabletDrawer from "./TabletDrawer"
import MobileBar from "./MobileBar"
import PropTypes from "prop-types"
const useStyles = makeStyles(theme => ({
  responsivePadding: {
    paddingLeft: 0,
    paddingBottom: theme.spacing(8),
    [theme.breakpoints.up(`sm`)]: {
      paddingLeft: theme.spacing(10) + 1,
      paddingBottom: 0,
    },
    [theme.breakpoints.up(`md`)]: {
      paddingLeft: 0,
    },
  },
}), { name: `lib-react-main_bar` })
export default function MainBar (props) {
  const classes = useStyles()
  return (
    <div className={classes.responsivePadding}>
      {useResponsiveLayout()({
        tablet: () => <TabletDrawer />,
        mobile: () => props.hideMobile ? null : <MobileBar />,
      })}
      {props.children}
    </div>
  )
}
MainBar.propTypes = {
  children: PropTypes.node,
  hideMobile: PropTypes.bool,
}