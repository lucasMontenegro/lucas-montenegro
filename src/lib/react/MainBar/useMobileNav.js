import { makeStyles } from "@material-ui/core/styles"
import React, { useState } from "react"
import Drawer from "@material-ui/core/Drawer"
import AppBar from "@material-ui/core/AppBar"
import Toolbar from "@material-ui/core/Toolbar"
import Tooltip from "@material-ui/core/Tooltip"
import IconButton from "@material-ui/core/IconButton"
import { FontAwesomeIcon } from "lib/react/fontAwesome"
import List from "@material-ui/core/List"
import { useRoutingContext } from "lib/react/routing/context"
import Link from "lib/react/links/Link"
import ListItem from "@material-ui/core/ListItem"
import ListItemIcon from "@material-ui/core/ListItemIcon"
import ListItemText from "@material-ui/core/ListItemText"
const useStyles = makeStyles(theme => ({
  drawerPaper: {
    paddingTop: theme.spacing(6),
    paddingBottom: theme.spacing(10),
  },
  appBar: {
    color: theme.palette.text.primary,
    backgroundColor: theme.palette.background.paper,
    top: `auto`,
    bottom: 0,
  },
  toolbar: {
    justifyContent: `flex-end`,
    "& > button.icon-button": {
      minWidth: theme.spacing(6),
    },
    "& > div.flexible-space": {
      flexBasis: `100%`,
      flexShrink: 1,
    },
  },
  li: {
    color: theme.palette.primary.main,
  },
  activeLi: {
    color: theme.palette.type === `dark` ? `#33ccd7` : `#017982`,
    backgroundColor: theme.palette.type === `dark` ? `#272727` : `#f1f1f1`,
  },
  anchor: {
    color: `inherit`,
    display: `block`,
  },
  liIcon: {
    color: `inherit`,
    fontSize: theme.spacing(3),
    minWidth: theme.spacing(5),
    marginRight: theme.spacing(2),
  },
}), { name: `lib-react-main_bar-mobile_nav` })
function MobileNav (props) {
  const classes = useStyles()
  return (
    <Drawer
      variant="temporary"
      anchor="bottom"
      id="lib-react-main_bar-mobile_nav"
      open={props.isOpen}
      onClose={props.close}
      classes={{ paper: classes.drawerPaper }}
    >
      <AppBar position="absolute" className={classes.appBar}>
        <Toolbar className={classes.toolbar}>
          <div className="flexible-space" />
          <Tooltip
            title={props.t({
              en: () => `Close quick navigation`,
              es: () => `Cerrar navegaci칩n r치pida`,
            })}
          >
            <IconButton
              edge="end"
              className="icon-button"
              aria-controls="lib-react-main_bar-mobile_nav"
              onClick={props.close}
            >
              <FontAwesomeIcon icon={[`fas`, `caret-down`]} />
            </IconButton>
          </Tooltip>
        </Toolbar>
      </AppBar>
      <nav
        aria-label={props.t({
          en: () => `Navigation links`,
          es: () => `Links de navegaci칩n`,
        })}
      >
        <List disablePadding>
          {useRoutingContext().clientLinks.map(link => (
            <li
              key={link.clientName}
              className={link.active ? `${classes.li} ${classes.activeLi}` : classes.li}
              aria-label={link.active ? props.t({
                en: () => `Current page`,
                es: () => `P치gina actual`,
              }) : undefined}
            >
              <Link className={classes.anchor} to={link.location} onClick={props.close}>
                <ListItem component="div">
                  <ListItemIcon className={classes.liIcon}>
                    <link.render.Icon t={props.t} />
                  </ListItemIcon>
                  <ListItemText>{props.t(link.render.text)}</ListItemText>
                </ListItem>
              </Link>
            </li>
          ))}
        </List>
      </nav>
    </Drawer>
  )
}
export default function useMobileNav (t) {
  const [isOpen, setOpenTo] = useState(false)
  return {
    node: <MobileNav t={t} isOpen={isOpen} close={() => setOpenTo(false)} />,
    open: () => setOpenTo(true),
  }
}