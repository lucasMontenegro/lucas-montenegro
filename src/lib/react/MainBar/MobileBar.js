import { makeStyles } from "@material-ui/core/styles"
import useTranslation from "lib/react/useTranslation"
import useMobileNav from "./useMobileNav"
import React from "react"
import AppBar from "@material-ui/core/AppBar"
import Toolbar from "@material-ui/core/Toolbar"
import Tooltip from "@material-ui/core/Tooltip"
import IconButton from "@material-ui/core/IconButton"
import { useAppDrawerOpener } from "lib/react/AppDrawer"
//import { useDashboardOpener } from "lib/react/Dashboard"
import { FontAwesomeIcon } from "lib/react/fontAwesome"
const useStyles = makeStyles(theme => ({
  appBar: {
    color: theme.palette.text.primary,
    backgroundColor: theme.palette.background.paper,
    top: `auto`,
    bottom: 0,
  },
  toolbar: {
    "& > button.icon-button": {
      minWidth: theme.spacing(6),
    },
    "& > div.flexible-space": {
      flexBasis: `100%`,
      flexShrink: 1,
    },
  },
}), { name: `lib-react-main_bar-mobile_bar` })
export default function MobileBar () {
  const classes = useStyles()
  const t = useTranslation()
  const mobileNav = useMobileNav(t)
  return (
    <AppBar className={classes.appBar}>
      <Toolbar className={classes.toolbar}>
        <Tooltip
          title={t({
            en: () => `Open dashboard`,
            es: () => `Abrir panel de control`,
          })}
        >
          <IconButton
            className="icon-button"
            edge="start"
            aria-controls="lib-react-app_drawer"
            onClick={useAppDrawerOpener()}
          >
          {/*
            aria-controls="lib-react-dashboard"
            onClick={useDashboardOpener()}
          >
          */}
            <FontAwesomeIcon icon={[`fas`, `bars`]} />
          </IconButton>
        </Tooltip>
        <div className="flexible-space" />
        <Tooltip
          title={t({
            en: () => `Open quick navigation`,
            es: () => `Abrir navegación rápida`,
          })}
        >
          <IconButton
            edge="end"
            className="icon-button"
            aria-controls="lib-react-main_bar-mobile_nav"
            onClick={mobileNav.open}
          >
            <FontAwesomeIcon icon={[`fas`, `caret-up`]} />
          </IconButton>
        </Tooltip>
        {mobileNav.node}
      </Toolbar>
    </AppBar>
  )
}