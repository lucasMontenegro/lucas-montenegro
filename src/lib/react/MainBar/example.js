import React, { useState } from "react"
import { FontAwesomeIcon } from "lib/react/fontAwesome"
import { makeStyles } from "@material-ui/core/styles"
import AppBar from "@material-ui/core/AppBar"
import Toolbar from "@material-ui/core/Toolbar"
import Typography from "@material-ui/core/Typography"
import MainBar from "./index.js"
import Div from "lib/react/utils/Div"
import GetValue from "lib/react/utils/GetValue"
import languageDetector from "lib/languageDetector"
import { DarkModeContext } from "lib/react/DarkMode"
import Theme from "lib/react/Theme"
import CssBaseline from "lib/react/CssBaseline"
import { RoutingContext } from "lib/react/routing/context"
import { AppDrawerContext } from "lib/react/AppDrawer"
import { Route } from "react-router-dom"
const languageCodes = [`en`, `es`]
const clients = { foo: `Foo`, bar: `Bar`, baz: `Baz` }
const baseLinks = [
  {
    clientName: `foo`,
    render: {
      Icon: props => props.t({
        en: () => <FontAwesomeIcon icon={[`fas`, `dog`]} />,
        es: () => <FontAwesomeIcon icon={[`fas`, `hippo`]} />,
      }),
      text: { en: () => `English Foo`, es: () => `Spanish Foo` },
    },
  },
  {
    clientName: `bar`,
    render: {
      Icon: props => props.t({
        en: () => <FontAwesomeIcon icon={[`fas`, `cat`]} />,
        es: () => <FontAwesomeIcon icon={[`fas`, `crow`]} />,
      }),
      text: { en: () => `English Bar`, es: () => `Spanish Bar` },
    },
  },
  {
    clientName: `baz`,
    render: {
      Icon: props => props.t({
        en: () => <FontAwesomeIcon icon={[`fas`, `horse`]} />,
        es: () => <FontAwesomeIcon icon={[`fas`, `frog`]} />,
      }),
      text: { en: () => `English Baz`, es: () => `Spanish Baz` },
    },
  },
]
const useStyles = makeStyles(theme => ({
  padding: {
    [theme.breakpoints.up(`sm`)]: {
      paddingTop: 64,
    },
    [theme.breakpoints.up(`md`)]: {
      paddingLeft: 256,
    },
  },
  appBar: {
    top: `auto`,
    right: 0,
    bottom: 0,
    left: `auto`,
    width: `100vw`,
    [theme.breakpoints.up(`sm`)]: {
      top: 0,
      bottom: `auto`,
      width: `calc(100vw - ${theme.spacing(10) + 1}px)`,
    },
    [theme.breakpoints.up(`md`)]: {
      width: `calc(100vw - ${theme.spacing(32) + 1}px)`,
    },
  },
}), { name: `lib-react-main_bar-example` })
function CustomBar (props) {
  const classes = useStyles()
  return (
    <div className={classes.padding}>
      {props.show && (
        <AppBar className={classes.appBar}>
          <Toolbar><Typography>Custom Bar</Typography></Toolbar>
        </AppBar>
      )}
      {props.children}
    </div>
  )
}
function Client (props) {
  const [isBarShown, showBar] = useState(false)
  return (
    <MainBar hideMobile={isBarShown}>
      <CustomBar show={isBarShown}>
        <Div>{clients[props.name]}</Div>
        <GetValue
          id="show-custom-bar"
          value={isBarShown}
          onClick={() => showBar(!isBarShown)}
        />
      </CustomBar>
    </MainBar>
  )
}
function Example (props) {
  languageDetector.init(languageCodes)
  if (languageDetector.useReadyState()) {
    const { mode, languageCode, currentClient } = props.match.params
    languageDetector.set(languageCode)
    return (
      <DarkModeContext.Provider value={{ value: mode === `dark` }}>
        <Theme>
          <CssBaseline />
          <RoutingContext.Provider
            value={{
              clientLinks: baseLinks.map(link => {
                const { clientName } = link
                return {
                  clientName,
                  active: clientName === currentClient,
                  render: link.render,
                  location: { pathname: `/react/MainBar/${mode}/${languageCode}/${clientName}` },
                }
              }),
            }}
          >
            <AppDrawerContext.Provider value={() => console.log(`Open App drawer`)}>
              <Client name={currentClient} />
            </AppDrawerContext.Provider>
          </RoutingContext.Provider>
        </Theme>
      </DarkModeContext.Provider>
    )
  }
  return null
}
export default (
  <Route exact path="/react/MainBar/:mode/:languageCode/:currentClient" component={Example} />
)