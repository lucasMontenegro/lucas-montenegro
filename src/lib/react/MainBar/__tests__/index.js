import React from "react"
import MainBar from "../index.js"
import renderer from "react-test-renderer"
jest.mock(`@material-ui/core/styles`, () => ({
  __esModule: true,
  makeStyles: (styles, options) => () => {
    const classes = styles({
      spacing: n => 8*n,
      breakpoints: { up: str => `theme.breakpoints.up('${str}')` },
    })
    return Object.keys(classes).reduce((obj, key) => {
      obj[key] = `${options.name}-${key}(${JSON.stringify(classes[key]).replace(/"/gm, `'`)})`
      return obj
    }, {})
  },
}))
jest.mock(`lib/react/useResponsiveLayout`, () => {
  const React = jest.requireActual("react")
  return {
    __esModule: true,
    default: () => source => (
      <div className="responsive-layout">
        {Object.keys(source).map(key => <div key={key} className={key}>{source[key]()}</div>)}
      </div>
    ),
  }
})
jest.mock(`../TabletDrawer`, () => {
  const React = jest.requireActual("react")
  return { __esModule: true, default: () => <div>TabletDrawer</div> }
})
jest.mock(`../MobileBar`, () => {
  const React = jest.requireActual("react")
  return { __esModule: true, default: () => <div>MobileBar</div> }
})
describe(`../index.js`, () => {
  it(`should use the right versions of its dependencies`, () => {
    expect(jestUtils.getDependencies([
      `@material-ui/core`,
      `react`,
      `prop-types`,
    ])).toMatchSnapshot()
  })
  describe.each([[true], [false]])(`<MainBar /> (hide mobile bar %j)`, hideMobile => {
    it(`should render`, () => {
      const node = <MainBar hideMobile={hideMobile}>children</MainBar>
      expect(renderer.create(node).toJSON()).toMatchSnapshot()
    })
  })
})