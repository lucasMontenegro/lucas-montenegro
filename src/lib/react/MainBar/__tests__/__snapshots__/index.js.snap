// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`../index.js <MainBar /> (hide mobile bar false) should render 1`] = `
<div
  className="lib-react-main_bar-responsivePadding({'paddingLeft':0,'paddingBottom':64,'theme.breakpoints.up('sm')':{'paddingLeft':81,'paddingBottom':0},'theme.breakpoints.up('md')':{'paddingLeft':0}})"
>
  <div
    className="responsive-layout"
  >
    <div
      className="tablet"
    >
      <div>
        TabletDrawer
      </div>
    </div>
    <div
      className="mobile"
    >
      <div>
        MobileBar
      </div>
    </div>
  </div>
  children
</div>
`;

exports[`../index.js <MainBar /> (hide mobile bar true) should render 1`] = `
<div
  className="lib-react-main_bar-responsivePadding({'paddingLeft':0,'paddingBottom':64,'theme.breakpoints.up('sm')':{'paddingLeft':81,'paddingBottom':0},'theme.breakpoints.up('md')':{'paddingLeft':0}})"
>
  <div
    className="responsive-layout"
  >
    <div
      className="tablet"
    >
      <div>
        TabletDrawer
      </div>
    </div>
    <div
      className="mobile"
    />
  </div>
  children
</div>
`;

exports[`../index.js should use the right versions of its dependencies 1`] = `
Object {
  "@material-ui/core": "^4.5.0",
  "prop-types": "^15.7.2",
  "react": "^16.10.1",
}
`;
